### PR TITLE
2017-2018 Honda CR-V, 2017 Honda Civic Hatchback, 2018 Honda Accord -- Lateral control + Stock ACC

### PR DESCRIPTION
**Pull request by energee**
Ready to be merged

### CODE REVIEW

**Comment by geohot** [comment]
Did you fix getting the radar objects?

**Comment by energee** [comment]
No there are no radar objects, and radar sends gas and brake command so we have to use stock ACC. 👎 Will demo longitudinal control without radar though if you are serious about making vision-only control work.

**Comment by willitblend** [comment]
I tested the stock ACC on my 2017 CR-V and I couldn't get it to cancel ACC using the accelerator. 

I set the Acc to 70 on the freeway
Got a car detected in front
Then used the accelerator to get closer and closer to the detected car


I got down to about 1 car length at 50mph and the Acc stayed on when I released the accelerator.

It's possible that I wasn't comfortable getting close enough to cancel the ACC, but it also would have been very unsafe to use any less following distance at freeway speeds.

In which conditions will the stock system cancel ACC on accelerator?

Had anyone experienced their ACC cancelling on accelerator?

Would it be possible to allow the visual Acc to override the stock radar acc if the camera detects an object the stock radar doesn't?

Or better, could it decelerate upon detection of a car which is out of radar range but stopped? This is an actual shortcoming of the stock ACC. The radar range is not long enough to leave the ACC engaged while approaching stopped traffic.

**Comment by 0x3333** [comment]
As long I as know, accelerating will not cancel ACC, brake will.

**Comment by davey1434** [comment]
I drive an 2018 CRV Touring; The 17 is the same... The Accell pedal does not cancel the ACC, it will just override throttle control and the ACC will resume throttle control when the pedal is released. It is possible to accel right into car in front if you really wanted to. The brake pedal, cancel switch, low traction event or over speed ( above 145KMH) will cancel the ACC operation. Hope the Giraffe will come out of this soon for the CRV /Bosch camera!

**Comment by cmiller49** [comment]
What else do I need to get going on my '17 CR-V /w sensing? I made a neo and 12p cable, and I just ordered a white panda. 
Are any other accessories required for lateral control? Without the panda, the EOS seems unable to calibrate or prevent lkas fault codes. Is this expected?

**Comment by willitblend** [comment]
Cmiller did you have any luck? I'm looking to get my 17 crv going but none had written it up. Someone put up a diagram for a 12p giraffe in the other thread and said it would be for sale 'soon'. I don't understand enough about all of this to get hacking yet but if you got yours working, would you mind writing up a guide?

**Comment by energee** [comment]
You need an EON + Panda + Giraffe for this code, it requires 3 canbuses, NEO only has two

**Comment by willitblend** [comment]
Is there a 12p giraffe available, or is everyone fabricating their own based on the diagram in the other thread?

**Comment by cmiller49** [comment]
That's a negative. There's two people with proof of concept videos on
YouTube, but I haven't gotten around to emailing them yet.
Please copy me if you learn anything
Regards.

File: selfdrive/controls/lib/planner.py [hunk 1]
```diff
@@ -453,8 +453,8 @@ def update(self, CS, LoC, v_cruise_kph, user_distracted):
     events = []
     if self.model_dead:
       events.append(create_event('modelCommIssue', [ET.NO_ENTRY, ET.IMMEDIATE_DISABLE]))
-    if self.radar_dead or 'commIssue' in self.radar_errors:
-      events.append(create_event('radarCommIssue', [ET.NO_ENTRY, ET.IMMEDIATE_DISABLE]))
+    # if self.radar_dead or 'commIssue' in self.radar_errors:
+    #   events.append(create_event('radarCommIssue', [ET.NO_ENTRY, ET.IMMEDIATE_DISABLE]))
```
Comment (energee) [inline selfdrive/controls/lib/planner.py]: @rickbias The radar interfaces are specific to each manufacturer's directory, but we have a different radar, we're also seeing this on the toyotas where some have continentals. Would it make sense to have a radar type in interface.py?

**Comment by achen415** [comment]
Just purchased an EON + Panda + Giraffe for my 2018 Honda CRV EX-L with sensing. Any one near San Francisco that could help me set up the Giraffe connector? I've read the previous post but don't have any experience with hacking cars. Not confident enough to do it myself. Will compensate generously for your time.

**Comment by energee** [comment]
@achen415 Not compatible with comma giraffe

**Comment by gregjhogan** [comment]
Since a giraffe isn't available for this vehicle yet, here is what you need to build to run this code.  Jumper wires soldered to an OBDII connector work great (male jumpers for wire harness side, female jumpers for camera side)

```
Panda CAN 2 → ACC CAN (camera 6/12)
Panda CAN 3 → ACC CAN (harness 6/12)
F-CAN B (harness 5/11) → Panda CAN1 → F-CAN B (camera 5/11)
IGN/GND (harness 1/2) → Panda IGN/GND → IGN/GND (camera 1/2)
LWD SW/IND (harness 7/8) → LWD SW/IND (camera 7/8)
```

**Comment by billy2kool** [comment]
Question how is it there's so much info on newer cars and not much on 2016
Honda Accord touring.  I am new to this and don't know anything but no one
seems to know about this car

**Comment by energee** [comment]
@billy2kool If you have questions about a particular car, please post on slack, not off-topic on other car threads.

**Comment by csouers** [comment]
@Geohot already threatened him with a ban because of this 1-2 weeks ago. He’s gone and done it again.

**Comment by willitblend** [comment]
I have reservations about the stock acc as the sole forward control. I often don't have the heart to let the car continue breaking when the free way stops or when the driver in front of me is reacting slowly to a red light. Do those of you using this set a really high follow distance?

**Comment by energee** [comment]
@willitblend Radar does not detect stopped cars well for technical reasons, this is common in most adaptive cruise control systems, including openpilot but if the system has detected the car in front of you it should be ok to stop safely.

File: opendbc/generator/honda/honda_accord_s2t_2018_can.dbc [hunk 2]
```diff
@@ -11,6 +11,12 @@ BO_ 446 BRAKE_MODULE: 3 VSA
  SG_ COUNTER : 21|2@0+ (1,0) [0|3] "" XXX
  SG_ CHECKSUM : 19|4@0+ (1,0) [0|15] "" XXX
 
+BO_ 662 SCM_BUTTONS: 8 SCM
```
Comment (rbiasini) [inline opendbc/generator/honda/honda_accord_s2t_2018_can.dbc]: @energee , this shouldn't be here as it's already in _bosch_2018.dbc.
Wondering why in the generated file 662 isn't repeated twice.

Comment (rbiasini) [reply opendbc/generator/honda/honda_accord_s2t_2018_can.dbc]: Also, the VAL line should be moved from the individual dbc file portions to here.

Comment (energee) [reply opendbc/generator/honda/honda_accord_s2t_2018_can.dbc]: Yea, that should be correct in opendbc, at one point i thought this had a different packet size on accord so i pulled  it out of the bosch dbc, but I must have been looking quickly at another bus, it matches the other cars. I will remerge the subtree. Did you want me adjust the VAL lines on OpenDBC?

**Comment by rbiasini** [comment]
@energee did we make the same comment at the same time? :)
Anyway, not clear why this happens but should be easy to find out. Let me know if you get stuck

**Comment by energee** [comment]
Haha, i realized you asked same thing right after i published. It looks like its trying to put an 8 into the counter singal on 0x296. I don't know too much about the test scripts but trying to catch myself up.

File: panda/board/drivers/can.h [hunk 3]
```diff
@@ -86,15 +86,15 @@ int can_err_cnt = 0;
   uint8_t can_num_lookup[] = {0,1,2,-1};
   int8_t can_forwarding[] = {-1,-1,-1,-1};
   uint32_t can_speed[] = {5000, 5000, 5000, 333};
-  bool can_autobaud_enabled[] = {false, false, false, false};
```
Comment (rbiasini) [inline panda/board/drivers/can.h]: @energee , I can remove those changes, right?
we don't need autobaud, do we?

Comment (energee) [reply panda/board/drivers/can.h]: Yes, this runs perfectly on RELEASE panda code.

File: README.md [hunk 4]
```diff
@@ -22,13 +22,24 @@ Supported Cars
 - Acura ILX 2016 with AcuraWatch Plus
   - Due to use of the cruise control for gas, it can only be enabled above 25 mph
 
+- Honda Accord 2018 with Honda Sensing (alpha!)
+  - Uses stock Honda Sensing for longitudinal control
```
Comment (rbiasini) [inline README.md]: This is true as well, right? @energee 
- Due to limitations in steering firmware, steering is disabled below 12 mph

Comment (energee) [reply README.md]: I don't think so, I believe accord steers down to 0.

Comment (mcampa) [reply README.md]: The Accord steers down to around 1.2 m/s. Below that number `STEER_STATUS` changes to 3 

image

File: selfdrive/car/honda/carstate.py [hunk 5]
```diff
@@ -275,18 +302,36 @@ def update(self, cp):
       self.steer_override = abs(cp.vl["STEER_STATUS"]['STEER_TORQUE_SENSOR']) > 1200
     self.steer_torque_driver = cp.vl["STEER_STATUS"]['STEER_TORQUE_SENSOR']
 
-    # brake switch has shown some single time step noise, so only considered when
-    # switch is on for at least 2 consecutive CAN samples
     self.brake_switch = cp.vl["POWERTRAIN_DATA"]['BRAKE_SWITCH']
-    self.brake_pressed = cp.vl["POWERTRAIN_DATA"]['BRAKE_PRESSED'] or \
+    if self.CP.safetyModel == car.CarParams.SafetyModels.hondaBosch:
+      self.stopped = cp.vl["ACC_HUD"]['CRUISE_SPEED'] == 252.
```
Comment (mcampa) [inline selfdrive/car/honda/carstate.py]: This probably needs to be renamed to `CRUISE_STATUS` IMO

Comment (energee) [reply selfdrive/car/honda/carstate.py]: Good point, will talk with @rbiasini about this, naming matches nidec hondas but i think this message matches them

**Comment by rbiasini** [comment]
Yeah, I would read the stopped state from the cruise state msg instead of
the hud.

File: selfdrive/car/honda/carstate.py (hunk 5 above)
Comment (rbiasini) [reply selfdrive/car/honda/carstate.py]: Yes, I would read the stopped state from a cruise state msg instead of the hud msg.

Comment (energee) [reply selfdrive/car/honda/carstate.py]: This is def a weird way to do this, but I can't seem to find another "car stopped" state anywhere else other than that message.

Comment (gregjhogan) [reply selfdrive/car/honda/carstate.py]: I also check `CS.hud_lead == 1.0` but I don't know if that really makes it any better

**Comment by energee** [comment]
Closing since it was merged internally by @rbiasini

**Pull request closed without merging**